app/src/main/java/Fragments/FavoriteFragment.java

```diff
@@ -102,49 +100,45 @@ public void onCancelled(DatabaseError databaseError) {
     }
 
 
-    public void populateFavoritesList(View view) {
-        if(!favList.isEmpty()) {
-            recyclerView = (RecyclerView) view.findViewById(R.id.fav_recycler_view);
-            mAdapter = new ListItemAdapter(listingsList);
-            RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(getActivity().getApplicationContext());
-            recyclerView.setLayoutManager(mLayoutManager);
-            recyclerView.setAdapter(mAdapter);
-            itemsRef.addValueEventListener(new ValueEventListener() {
-                @Override
-                public void onDataChange(DataSnapshot dataSnapshot) {
-                    listingsList.clear();
 
-                    Map<String, Object> itemsMap = (HashMap<String, Object>) dataSnapshot.getValue();
+    public void populateFavoritesList(View view, final DataSnapshot favRef) {
 
-                    for (String key : itemsMap.keySet()) {
-                        if (favList.contains(key)) {
-                            Object itemMap = itemsMap.get(key);
-                            if (itemMap instanceof Map) {
-                                Map<String, Object> itemObj = (Map<String, Object>) itemMap;
+        recyclerView = (RecyclerView) view.findViewById(R.id.fav_recycler_view);
+        mAdapter = new ListItemAdapter(listingsList);
+        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(getActivity().getApplicationContext());
+        recyclerView.setLayoutManager(mLayoutManager);
+        recyclerView.setAdapter(mAdapter);
 
-                                String id = key;
-                                String name = (String) itemObj.get("Name");
-                                Double price = ((Number) itemObj.get("Price")).doubleValue();
-                                String url = (String) itemObj.get("ImageURL");
+        itemsRef.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                listingsList.clear();
+                Map<String, Object> favMap = (HashMap<String, Object>) favRef.getValue();
+                Map<String, Object> itemsMap = (HashMap<String, Object>) dataSnapshot.getValue();
 
-                                Listing item = new Listing(key, name, price, url);
+                for (String key : favMap.keySet()) {
+                        Object itemMap = itemsMap.get(key);
+                        if (itemMap instanceof Map) {
+                            Map<String, Object> itemObj = (Map<String, Object>) itemMap;
 
-                                listingsList.add(item);
+                            String id = key;
```

Comment:
Don't forget to add the string filter handling. (i.e., if the user is selecting favourites, and then uses the search bar to search for something, he should be receiving the filtered favourites)

Reply:
Currently there's no search bar in the favorite fragment. Do we want to be able to search in favorites?

Reply:
Oh well if there's no search bar I guess it doesn't matter. This could be discussed but may be an unnecessarily annoying change to make.